### PR TITLE
Fix test tracking

### LINKED ISSUE #6694: test_tracking_integration - fix skipped tests
Make tests/unit_tests/status/test_integration_cli.py pass.

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves #6694 

**Approach**

Implement run_timeout_callback and run_exit_callback

### CODE REVIEW

File: src/ert/scheduler/scheduler.py [hunk 1]
```diff
@@ -347,6 +362,8 @@ async def execute(
                     await self._statechanges_to_publish.put(CLOSE_PUBLISHER_SENTINEL)
                     return EVTYPE_ENSEMBLE_CANCELLED
 
+                await asyncio.sleep(1)
```
Comment (berland) [inline src/ert/scheduler/scheduler.py:365]: Moved here to allow async tasks to get done before we test for active-ness.

File: src/ert/callbacks.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 from ert.run_arg import RunArg
 
 from .load_status import LoadResult, LoadStatus
-from .realization_state import RealizationState
+from .realization_state import RealizationState as RealizationStorageState
```
Comment (kwinkunks) [inline src/ert/callbacks.py:13]: I thought @xjules had renamed this already? Or I don't understand where `.` is here...

Comment (xjules) [reply src/ert/callbacks.py:13]: Yes, on main! Still need to todo a major rebase :)

File: src/ert/scheduler/realization_state.py [hunk 3]
```diff
@@ -123,17 +123,18 @@ def on_enter_RUNNING(self) -> None:
...
     def on_enter_EXIT(self) -> None:
-        self.realization.run_arg.ensemble_storage.state_map[
-            self.realization.run_arg.iens
-        ] = RealizationStorageState.LOAD_FAILURE
-
         if self.retries_left > 0:
             self.retry()
             self.retries_left -= 1
         else:
+            self.realization.run_arg.ensemble_storage.state_map[
```
Comment (xjules) [inline src/ert/scheduler/realization_state.py:130]: Shouldn't we change state to `DO_KILL` in here?

Comment (berland) [reply src/ert/scheduler/realization_state.py:130]: No, we don't need to kill a realization that has already exited.

File: src/ert/scheduler/scheduler.py [hunk 4]
```diff
@@ -124,18 +125,29 @@ async def run_done_callback(self, state: RealizationState) -> Optional[LoadStatu
             f"Running done callback for {state.realization.run_arg.iens} ..(partly blocking)..",
             end="",
         )
-        callback_status, state._callback_status_msg = await forward_model_ok(
-            state.realization.run_arg
-        )
+        callback_status, status_msg = await forward_model_ok(state.realization.run_arg)
         print(" done")
+
+        if state._callback_status_msg != "":
```
Comment (xjules) [inline src/ert/scheduler/scheduler.py:131]: Isn't this flipped?

File: src/ert/scheduler/scheduler.py [hunk 5]
```diff
@@ -124,18 +125,29 @@ async def run_done_callback(self, state: RealizationState) -> Optional[LoadStatu
...
         if callback_status == LoadStatus.LOAD_SUCCESSFUL:
             state.validate()
         elif callback_status == LoadStatus.TIME_MAP_FAILURE:
             state.invalidate()
         else:  # LoadStatus.LOAD_FAILURE
             state.loadfailed()
+
         return callback_status
 
+    async def run_timeout_callback(self, realization: RealizationState) -> None:
```
Comment (xjules) [inline src/ert/scheduler/scheduler.py:145]: We should unify calling this state and not realization I think.

**Review by xjules — CHANGES_REQUESTED** [review]
Got a few comments, but otherwise looks good!

File: src/ert/scheduler/scheduler.py (hunk 4 above)
Comment (berland) [reply src/ert/scheduler/scheduler.py:131]: This looks strange but is copied directly from the existing job_queue_node, and it must be this way to pass tests ;)

File: src/ert/scheduler/scheduler.py (hunk 5 above)
Comment (berland) [reply src/ert/scheduler/scheduler.py]: fixed

**Comment by berland** [comment]
Repaired erroneous force push

**Review by xjules — APPROVED** [review]
Nice work! :rocket:

**Pull request merged**